src/import/importConfiguration.ts

```diff
@@ -0,0 +1,713 @@
+import YAML from 'yaml';
+import {
+  YAMLConfiguration,
+  YAMLConfigurationPage,
+  YAMLConfigurationMenuTree,
+  YAMLConfigurationDependencyMap,
+  YAMLConfigurationGlobalMap,
+  YAMLConfigurationGroupMap,
+  YAMLConfigurationResourceMap,
+  YAMLConfigurationPageTree,
+  YAMLConfigurationMeta,
+  YAMLConfigurationUsageContext,
+  YAMLConfigurationQuantity,
+  YAMLConfigurationNarrative,
+  YAMLConfigurationRange,
+  YAMLConfigurationReference,
+  YAMLConfigurationIdentifier
+} from './YAMLConfiguration';
+import {
+  Configuration,
+  ConfigurationGroup,
+  ConfigurationResource,
+  ConfigurationMenuItem,
+  ConfigurationHistory,
+  ConfigurationHistoryItem
+} from '../fshtypes/Configuration';
+import { logger } from '../utils/FSHLogger';
+import { parseCodeLexeme } from './parseCodeLexeme';
+import {
+  ImplementationGuideGlobal,
+  ImplementationGuideDefinitionPage,
+  ImplementationGuideDefinitionParameter,
+  ContactDetail,
+  ImplementationGuideDependsOn,
+  Meta,
+  Coding,
+  Narrative,
+  ContactPoint,
+  CodeableConcept,
+  UsageContext,
+  Quantity,
+  Range,
+  Identifier,
+  Reference,
+  ImplementationGuideDefinitionTemplate
+} from '../fhirtypes';
+import { FshCode } from '../fshtypes';
+
+const MINIMAL_CONFIG_PROPERTIES = ['id', 'version', 'url', 'fhirVersion'];
+
+/**
+ * Imports the YAML Configuration format (as a YAML string or already parsed JSON) and returns
+ * the normalized FSH Configuration object.
+ * @param {YAMLConfiguration | string} yaml - the YAML config as a string or JSON document
+ * @returns {Configuration} - the FSH configuration representing the parsed config
+ */
+export function importConfiguration(yaml: YAMLConfiguration | string, file: string): Configuration {
+  if (typeof yaml === 'string') {
+    return importConfiguration(YAML.parse(yaml) as YAMLConfiguration, file);
+  }
+
+  // There are a few properties that are absolutely required if we are to have *any* success at all
+  if (
+    MINIMAL_CONFIG_PROPERTIES.some(
+      // @ts-ignore Element implicitly has an 'any' type
+      p => yaml[p] == null || (Array.isArray(yaml[p]) && yaml[p].length === 0)
+    )
+  ) {
+    logger.error(
+      'SUSHI minimally requires the following configuration properties to start processing FSH: ' +
+        MINIMAL_CONFIG_PROPERTIES.join(', ') +
+        '.',
+      { file }
+    );
+    throw new Error('Minimal config not met');
+  }
+
+  const config: Configuration = {
+    filePath: file,
+    id: yaml.id, // minimum config property
+    meta: parseMeta(yaml.meta, file),
+    implicitRules: yaml.implicitRules,
+    language: parseSimpleCode(yaml.language, 'language', file),
+    text: parseText(yaml.text, file),
+    contained: yaml.contained,
+    extension: yaml.extension,
+    modifierExtension: yaml.modifierExtension,
+    url: yaml.url, // minimum config property
+    version: normalizeToString(yaml.version), // minimum config property
+    name: required(yaml.name, 'name', file),
+    title: yaml.title,
```

Comment:
Yeah.  I think since it is technically `0..1` in `ImplementationGuide`, then we won't force it to be populated.  I'm trying to give authors a bit more control instead of forcing things.